protocol/shannon/fullnode_session_rollover.go

```diff
@@ -0,0 +1,167 @@
+package shannon
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	sessiontypes "github.com/pokt-network/poktroll/x/session/types"
+)
+
+const (
+	// TODO_IN_THIS_PR(@commoddity): Make this configurable via config file
+	// Grace period after session end where rollover issues may occur
+	sessionRolloverGracePeriodBlocks = 10
```

Comment:
I believe we are at 60 blocks per session, setting this to a more aggressive initial value of 5 seems to make more sense considering our goal here.

Reply:
Generally our observed errors during session rollover happen during a 5 minute period, so my estimate was:
` 10 blocks* 30 seconds = 5 minutes` 
as a starting point. Do you think this is incorrect?

Reply:
> Trying to minimize the use of fallback endpoints, hence the suggestion for a shorter window during which we use fallback.

My concern is that:
1. a shorter window will not cover the full period where we experience elevated error rates
2. this PR turns off ALL fallback outside this window

Therefore we have the risk of seeing error rates during session rollover go back up.

Going from using fallback 100% of the time to 1/6th of the time is already a huge reduction so IMO it's better to err on the side of caution and not risk error rates rising due to turning off fallback during a period where error rates may be higher than usual.